library/src/main/java/com/mux/video/upload/internal/UploadMetrics.kt

```diff
@@ -13,55 +13,73 @@ import okhttp3.Interceptor
 import okhttp3.MediaType.Companion.toMediaType
 import okhttp3.Request
 import okhttp3.RequestBody.Companion.toRequestBody
+import org.json.JSONArray
 import org.json.JSONObject
+import java.text.SimpleDateFormat
 import java.util.*
 import java.util.concurrent.TimeUnit
 
+
 internal class UploadMetrics private constructor() {
 
   private val logger get() = MuxUploadSdk.logger
 
-  @JvmSynthetic
-  internal suspend fun reportUpload(
-    startTimeMillis: Long,
-    endTimeMillis: Long,
-    uploadInfo: UploadInfo
-  ) {
-    val videoDuration = withContext(Dispatchers.IO) {
-      try {
-        MediaMetadataRetriever().use { retriever ->
-          retriever.setDataSource(uploadInfo.inputFile.absolutePath)
-          retriever.extractMetadata(MediaMetadataRetriever.METADATA_KEY_DURATION)?.toInt()
+
+  private fun formatMilliseconds(ms:Long):String {
+    return String.format("%02d:%02d:%02d",
+      TimeUnit.MILLISECONDS.toHours(ms),
+      TimeUnit.MILLISECONDS.toMinutes(ms) -
+              TimeUnit.HOURS.toMinutes(TimeUnit.MILLISECONDS.toHours(ms)), // The change is in this line
+      TimeUnit.MILLISECONDS.toSeconds(ms) -
+              TimeUnit.MINUTES.toSeconds(TimeUnit.MILLISECONDS.toMinutes(ms)));
+  }
+
+  private suspend fun getEventInfo(startTimeMillis: Long,
+                                   endTimeMillis: Long,
+                                   inputFileDurationMs: Long,
+                                   uploadInfo: UploadInfo): JSONObject {
+    val iso8601Sdf = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ssZ", Locale.getDefault())
+    var videoDuration = inputFileDurationMs
+    if (inputFileDurationMs <= 0) {
+      videoDuration = withContext(Dispatchers.IO) {
+        try {
+          MediaMetadataRetriever().use { retriever ->
+            retriever.setDataSource(uploadInfo.inputFile.absolutePath)
+            retriever.extractMetadata(MediaMetadataRetriever.METADATA_KEY_DURATION)?.toInt()
+          }
+        } catch (e: Exception) {
+          MuxUploadSdk.logger.e("UploadMetrics", "Failed to get video duration", e)
+          null
         }
-      } catch (e: Exception) {
-        MuxUploadSdk.logger.e("UploadMetrics", "Failed to get video duration", e)
-        null
-      }
+      }!!.toLong()
```

Comment:
Instead of `!!` you should use `?:`. Otherwise failing to extract the duration would stop the entire event from going up (or cause a crash), which is not necessary.

```suggestion
      } ?: 0L 
```